### ISSUE TITLE
Mysterious error when using pytest-pylint

### ISSUE THREAD

**Issue by PlasmaSheep** [carsongee/pytest-pylint#22]
Hello,

I seem to be getting weird behavior when using pylint on this repo:

https://github.com/PlasmaSheep/pylint-bug

(issue present in python2 and python3)

```
pip install -r requirements.txt
py.test --pylint
```

Will cause:

```

    def test_index():
>       create_app()

tests/test_views_core.py:7: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def create_app():
        """Create and return an instance of this application.
        """
>       from cacophony.views.core import core
E       ImportError: No module named 'cacophony.views.core'
```

Which clearly exists. However, `py.test` without pylint does not cause this error.

**Comment by carsongee** [comment]
I don't think this is pylint related directly:
```python
(rad) #[19:13:02] carson@cgee-lt00:~/tmp/pylint-bug $
ipython
Python 2.7.10 (default, Oct 23 2015, 19:19:21)
Type "copyright", "credits" or "license" for more information.

IPython 5.0.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from cacophony.cacophony import create_app

In [2]: create_app()
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-2-c0a18d81702c> in <module>()
----> 1 create_app()

/Users/carson/tmp/pylint-bug/cacophony/cacophony.pyc in create_app()
      4     """Create and return an instance of this application.
      5     """
----> 6     from cacophony.views.core import core

ImportError: No module named views.core

In [3]:
```
It appears to be more of a python module import bug related to relative imports. i.e. it can't find `cacophony.views.core` from `cacophony.cacophony`.  You should be able to fix with just changing the import statement in ` cacophony.cacophony` to be `from views.core import core`, i.e.:
```
git diff
diff --git a/cacophony/cacophony.py b/cacophony/cacophony.py
index f9006ca..ce23bca 100644
--- a/cacophony/cacophony.py
+++ b/cacophony/cacophony.py
@@ -3,4 +3,4 @@
 def create_app():
     """Create and return an instance of this application.
     """
-    from cacophony.views.core import core
+    from views.core import core
(rad) #[19:16:09] carson@cgee-lt00:~/tmp/pylint-bug $
ipython
Python 2.7.10 (default, Oct 23 2015, 19:19:21)
Type "copyright", "credits" or "license" for more information.

IPython 5.0.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from cacophony.cacophony import create_app

In [2]: create_app()

In [3]:
```

with that change I get:
```
py.test --pylint
===================================================================================== test session starts =====================================================================================
platform darwin -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /Users/carson/tmp/pylint-bug, inifile:
plugins: cov-2.3.0, pep8-1.0.6, pylint-0.6.0
collected 7 items

cacophony/__init__.py .
cacophony/cacophony.py F
cacophony/views/__init__.py .
cacophony/views/core.py F
tests/__init__.py .
tests/test_views_core.py F.

========================================================================================== FAILURES ===========================================================================================
_______________________________________________________________________________ [pylint] cacophony/cacophony.py _______________________________________________________________________________
W:  6, 4: Relative import 'views.core', should be 'cacophony.views.core' (relative-import)
W:  6, 4: Unused variable 'core' (unused-variable)
------------------------------------------------------------------------------------ Captured stderr call -------------------------------------------------------------------------------------
No config file found, using default configuration
______________________________________________________________________________ [pylint] cacophony/views/core.py _______________________________________________________________________________
C:  8, 0: Trailing newlines (trailing-newlines)
C:  6, 0: Invalid constant name "core" (invalid-name)
W:  4, 0: Unused render_template imported from flask (unused-import)
------------------------------------------------------------------------------------ Captured stderr call -------------------------------------------------------------------------------------
No config file found, using default configuration
______________________________________________________________________________ [pylint] tests/test_views_core.py ______________________________________________________________________________
C:  6, 0: Missing function docstring (missing-docstring)
------------------------------------------------------------------------------------ Captured stderr call -------------------------------------------------------------------------------------
No config file found, using default configuration
============================================================================= 3 failed, 4 passed in 0.39 seconds ==============================================================================
(rad) #[19:17:17] carson@cgee-lt00:~/tmp/pylint-bug $
```

**Comment by PlasmaSheep** [comment]
But if you do this, then pylint will throw the error:

```
W:  6, 4: Relative import 'views.core', should be 'cacophony.views.core' (relative-import)
```

I'm not sure if it's a python import bug, because it works without pytest-pylint and while running the application - so it seems that there is something related to pytest-pylint that is causing this (and possibly something in commmon with ipython).
